Sources/ApodiniDatabase/Handlers/Create.swift

```diff
@@ -0,0 +1,30 @@
+import Fluent
+import Apodini
+
+/// A Handler that creates, if possible, an object in the database that conforms to `DatabaseModel` out of the body of the request.
+/// It uses the database that has been specified in the `DatabaseConfiguration`.
+public struct Create<T: DatabaseModel>: Handler {
+    @Apodini.Environment(\.database)
+    private var database: Fluent.Database
+    
+    @Parameter
+    private var object: T
+
+    public func handle() -> T? {
+        // swiftlint:disable all
+        let result = try! object
+            .save(on: database)
+            .map { _ in
+                self.object
+            }
+            .wait()
+        
+        return try! T
```

Comment:
What is the reasoning behind fetching the type from the database? Would it be fine to store `object` to a local property and return it after saving the element to the database. I think as `object` is an instance of a class type the ID should also be updated after the save call.

Reply:
I think im not quite sure what you mean. This code is due `EventLoopFutures` not being a possible return type. Once this is possible again `Create` would return an `EventLoopFuture<Model>, where `Model` is the saved object. If we would store `object` to a local property and return it after the database save, we would again have some kind of blocking. But I agree with you that the returned object should have the saved id, I'll add that

Reply:
Sorry for the confusing, I could have been clearer with my comment. Here is an example of what I have meant with leaving out the fetching the type from the database.

Fluent automatically updates the ID property on the save call. So you could have a code similar to this one that does both the saving and the returning the object in a single call as seen in the following code block:
```swift
object
    .save(on: database)
    .transform(to: object)
    .wait()
```

Once we have merged #94 or #98 or a mixture of both you can simply remove the `wait()` call and the method should return the instance include the ID that has been created from the database.

[The Vapor template does something similar but uses `map` instead of `transform`](https://github.com/vapor/template/blob/0f3b552f66aee092b457a542d9a83dd5578d05b6/Sources/App/Controllers/TodoController.swift#L20) which has the same effect without the need for a closure in this context.

Reply:
Thanks for clarifying! I agree, thats looks a lot cleaner. I will update the `Create` Handler accordingly.